scRNA-seq-advanced/setup/intro-diff-expression-setup.Rmd

```diff
@@ -0,0 +1,633 @@
+
+```{r}
+# mean difference plots
+purrr::walk(1:ncol(dge_list), 
+            ~ limma::plotMD(dge_list, column = .x))
+```
+
+```{r}
+# MDS scaling plot 
+# first compute counts per million 
+counts_per_million <- edgeR::cpm(dge_list, log = TRUE)
+
+dge_list$samples <- dge_list$samples %>%
+  dplyr::mutate(diagnosis_groups = dplyr::case_when(
+    subdiagnosis == "Alveolar rhabdomyosarcoma" ~ "ARMS",
+    subdiagnosis == "Embryonal rhabdomyosarcoma" ~ "ERMS"
+  ))
+
+limma::plotMDS(counts_per_million,
+        # add some colors based on our two groups
+        col = ifelse(dge_list$samples$diagnosis_groups == "ARMS", "black", "red"))
+```
+
+Let's do some DE now! 
+
+```{r}
+# set up design matrix to compare between ARMS and ERMS groups
+design_matrix <- model.matrix(~factor(subdiagnosis),
+                              dge_list$samples)
+design_matrix
+```
+
+Estimate dispersions using both the quasi-likelihood(QL) dispersion and the negative binomial(NB) dispersions. 
+
+```{r}
+# estimate NB dispersions 
+dge_list <- edgeR::estimateDisp(dge_list, design = design_matrix)
+summary(dge_list$trended.dispersion)
+```
+
+```{r}
+edgeR::plotBCV(dge_list)
+```
+
+```{r}
+# estimate the quasi-likelihood dispersions
+glm_fit <- edgeR::glmQLFit(dge_list, design_matrix, robust = TRUE)
+edgeR::plotQLDisp(glm_fit)
+```
+
+Test for differences using the `glmQLFTest` where differentially expressed gens are defined as those with non-zero LFC and a FDR of < 5%.
+
+```{r}
+results <- edgeR::glmQLFTest(glm_fit, coef = ncol(design_matrix))
+summary(limma::decideTests(results))
+```
+
+```{r}
+edger_results <- edgeR::topTags(results, n="Inf")$table
+edger_results_sig <- edger_results %>%
+  dplyr::filter(FDR <= 0.05)
+
+# print out results 
+edger_results_sig
+```
+
+### Perform differential expression with DESeq2
+
+Let's do the same thing, but try with DESeq2.
+
+```{r}
+# deseq2 requires the coldata object as a separate input
+coldata_df <- as.data.frame(colData(tumor_myoblast_sce)) %>%
+  dplyr::mutate(diagnosis_groups = dplyr::case_when(
+    subdiagnosis == "Alveolar rhabdomyosarcoma" ~ "ARMS",
+    subdiagnosis == "Embryonal rhabdomyosarcoma" ~ "ERMS"
+  ),
+  diagnosis_groups = as.factor(diagnosis_groups)) 
+
+# set the levels
+coldata_df$diagnosis_groups <- relevel(coldata_df$diagnosis_groups, ref = "ARMS")
+  
+# set up the deseq object 
+deseq_object <- DESeq2::DESeqDataSetFromMatrix(counts(tumor_myoblast_sce),
+                                               colData = coldata_df,
+                                               design = ~ diagnosis_groups)
+```
+
+Use the median of ratios method for count normalization followed by regularized log transformation.
+
+```{r}
+# we get a warning if we don't use local that it can't find a model
+# this could likely be due to batch effects present in the data
+normalized_object <- DESeq2::rlog(deseq_object, blind = TRUE, fitType = 'local')
```

Comment:
I just tried that and I get the same error message 

> -- note: fitType='parametric', but the dispersion trend was not well captured by the
   function: y = a/x + b, and a local regression fit was automatically substituted.
   specify fitType='local' or 'mean' to avoid this message next time.